scalafix-rules/src/main/scala/scalafix/internal/rule/RemoveUnusedConfig.scala

```diff
@@ -16,7 +16,9 @@ case class RemoveUnusedConfig(
       "Remove unused pattern match variables (compatible with Scala 2.12 and 2.13)"
     )
     patternvars: Boolean = true,
-    @Description("Remove unused parameters")
+    @Description(
+      "Remove unused function parameters (compatible with Scala 2.12 and 2.13)"
```

Comment:
Unused method parameters are reported by the compiler but not handled by the rule, so I decided to make this explicit in the doc.

Since `def fn(_: Long)` is invalid syntax, we would have to change the signature, which is not safe (or would be limited to private methods and require quite a bit of logic for patching call sites). Therefore I don't anticipate the rule to ever support them, but if we ever do, I guess we can add a `methodparams` config flag (in other word, I think it's acceptable to assign `params` for function params only).

Reply:
Thank you very much for clarifying this.
I'd agree with that this RemoveUnused rule should not support unused explicit method parameters because of unsafety (that's why I have not added) and be nicer to be explicit about it.

One thing I was wondering is about unused implicit method parameters as an opt-in config.
Of course they are also not 100% safe to remove (because call sites can pass the argument explicitly), but still useful in most cases, and easy to fix even if it breaks the code.

```scala
/* O is not used */
def f[T](t: List[T])(implicit O: Ordering[T]): List[T] =  t

// It's safe to remove O
f(List("a"))

// It's not safe to remove O, although this is not so often.
f(List(1))(Ordering.Int)
```

Reply:
> I'd like to just note that I was wondering about unused implicit method parameters as an opt-in config.

Then we agree: a potential new opt-in flag could be added later, but we keep this current one for functions only.